src/components/CreateNFT.vue

```diff
@@ -0,0 +1,454 @@
+
+@Component({
+  components: {
+    InfoLine,
+    WalletFee,
+    NftDetails,
+    NetworkFeeWarningDialog,
+  },
+})
+export default class CreateNFT extends Mixins(
+  TranslationMixin,
+  TransactionMixin,
+  LoadingMixin,
+  NumberFormatterMixin,
+  NetworkFeeWarningMixin
+) {
+  readonly tokenSymbolMask = 'AAAAA';
+  readonly tokenNameMask = { mask: 'Z*', tokens: { Z: { pattern: /[0-9a-zA-Z ]/ } } };
+  readonly maxTotalSupply = MaxTotalSupply;
+  readonly decimals = 0;
+  readonly delimiters = FPNumber.DELIMITERS_CONFIG;
+  readonly Step = Step;
+  readonly XOR = XOR;
+
+  @Action navigate!: (options: { name: string; params?: object }) => Promise<void>;
+
+  @Prop({ default: Step.CreateToken, type: String }) step!: Step;
+
+  imageLoading = false;
+  badSource = false;
+  contentSrcLink = '';
+  tokenContentIpfsParsed = '';
+  tokenContentLink = '';
+  tokenSymbol = '';
+  tokenName = '';
+  tokenDescription = '';
+  tokenSupply = '';
+  linkPlaceholder = this.t('createToken.nft.link.placeholder');
+  showFee = true;
+  file: any = null;
+  fileName = '';
+
+  get isCreateDisabled(): boolean {
+    return (
+      !(this.tokenSymbol && this.tokenName.trim() && this.tokenSupply && this.tokenDescription.trim()) || this.badSource
+    );
+  }
+
+  get fee(): FPNumber {
+    return this.getFPNumberFromCodec(this.networkFees.RegisterAsset);
+  }
+
+  get formattedFee(): string {
+    return this.fee.toLocaleString();
+  }
+
+  get formattedFileName(): string {
+    return shortenFileName(this.fileName);
+  }
+
+  get contentSource(): string {
+    if (this.file) return this.t('createToken.nft.source.value');
+    return getUrlContentSource(this.tokenContentLink);
+  }
+
+  get hasEnoughXor(): boolean {
+    const accountXor = api.accountAssets.find((asset) => asset.address === XOR.address);
+    if (!accountXor || !accountXor.balance || !+accountXor.balance.transferable) {
+      return false;
+    }
+    const fpAccountXor = this.getFPNumberFromCodec(accountXor.balance.transferable, accountXor.decimals);
+    return FPNumber.gte(fpAccountXor, this.fee);
+  }
+
+  async handleFileUpload(): Promise<void> {
+    this.imageLoading = true;
+    const fileInput = this.$refs.fileInput as HTMLInputElement;
+    const file = (fileInput.files as FileList)[0];
+    if (!file) {
+      this.resetFileInput();
+      return;
+    }
+    this.file = file;
+    this.fileName = file.name;
+    this.contentSrcLink = (await fileToBase64(file)) as string;
+    this.badSource = false;
+    this.imageLoading = false;
+    this.tokenContentLink = '';
+  }
+
+  handleInputLinkChange(link: string): void {
+    this.imageLoading = true;
+    this.badSource = false;
+    this.linkPlaceholder = link
+      ? this.t('createToken.nft.link.placeholderShort')
+      : this.t('createToken.nft.link.placeholder');
+
+    this.checkImageFromSource(link);
+  }
+
+  async checkImageFromSource(url: string): Promise<void> {
```

Comment:
I've tried to upload a random image by link, but access is blocked by CORS. Looks like we should handle this error, and stop showing loader inside dropbox area

![nft-upload-link](https://user-images.githubusercontent.com/53777036/149278767-f795c313-6659-4150-b468-e78fc6079ef8.jpg)

Reply:
Fixed. Wrapped in try/catch block